File: javascript/anagram/anagram.js
function Anagram(word) {
    "use strict";
    
    var normalizedWord = normalize(word);

    function isAnagram(candidate) {
        return normalize(candidate) === normalizedWord;
    }

    function normalize(str) {
        return str.split("").sort().join("");
    }

    return {
        match: function(candidates) {
            return candidates.filter(isAnagram);
        }
    };    
};
module.exports = Anagram;

File: clojure/anagram/anagram.clj
(ns anagram
  (:require [clojure.string :as str]))

(defn- word-char-frequencies [word]
  (frequencies (str/upper-case word)))

(defn anagrams-for [word, candidates]
  (let [norm-word (word-char-frequencies word)] 
    (filter 
      (fn [candidate] 
        (= norm-word) (word-char-frequencies candidate)))
      candidates)))